xcube/core/compute.py

```diff
@@ -110,41 +162,53 @@ def cube_func(*input_vars: np.ndarray,
     elif input_cube_schema is None:
         raise ValueError('input_cube_schema must be given')
 
-    if output_var_name is None:
-        output_var_name = 'output'
+    output_var_name = output_var_name or 'output'
 
+    # Collect named input variables, raise if not found
     input_var_names = input_var_names or []
     input_vars = []
     for var_name in input_var_names:
-        var = None
+        input_var = None
         for cube in input_cubes:
             if var_name in cube.data_vars:
-                var = cube[var_name]
+                input_var = cube[var_name]
                 break
-        if var is None:
+        if input_var is None:
             raise ValueError(f'variable {var_name!r} not found in any of cubes')
-        input_vars.append(var)
+        input_vars.append(input_var)
 
+    # Find out, if cube_func uses any of _PREDEFINED_KEYWORDS
     has_input_params, has_dim_coords, has_dim_ranges = _inspect_cube_func(cube_func, input_var_names)
 
     def cube_func_wrapper(index_chunk, *input_var_chunks):
         nonlocal input_cube_schema, input_var_names, input_params, input_vars
         nonlocal has_input_params, has_dim_coords, has_dim_ranges
 
-        index_chunk = index_chunk.ravel()
+        # Note, xarray.apply_ufunc does a test call with empty input arrays,
+        # so index_chunk.size == 0 is a valid case
+        empty_call = index_chunk.size == 0
+
+        # TODO: when output_var_dims is given, index_chunk must be reordered
+        #   as core dimensions are moved to the and of index_chunk and input_var_chunks
+        if not empty_call:
+            index_chunk = index_chunk.ravel()
 
         if index_chunk.size < 2 * input_cube_schema.ndim:
-            warnings.warn(f"weird index_chunk of size {index_chunk.size} received!")
-            return
+            if not empty_call:
+                warnings.warn(f"weird index_chunk of size {index_chunk.size} received!")
```

Comment:
I thinkt this sounds better - but as you like.

```suggestion
                warnings.warn(f"unexpected index_chunk of size {index_chunk.size} received!")
```